### PR TITLE
En 3880 integration tests create test sync node

### PR DESCRIPTION
**Pull request by iulianpascalau**
added possibility for the testProcessorNode to have sync capabilities
added new constructor that creates a new sync-capable testProcessorNode
added first sync integration test with empty blocks

### CODE REVIEW

File: integrationTests/sync/basicSync_test.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	defer func() {
+		_ = advertiser.Close()
+		for _, n := range nodes {
+			_ = n.Messenger.Close()
+		}
+	}()
+
+	for _, n := range nodes {
+		_ = n.Messenger.Bootstrap()
+		_ = n.StartSync()
```
Comment (bogdan-rosianu) [inline integrationTests/sync/basicSync_test.go]: maybe treat the errors and if they aren't nil, print them and continue the range

File: integrationTests/sync/basicSync_test.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	metachainNode := integrationTests.NewTestSyncNode(
+		maxShards,
+		sharding.MetachainShardId,
+		shardId,
+		advertiserAddr,
+	)
+	nodes[numNodesPerShard] = metachainNode
+
+	idxProposerShard0 := 0
+	idxProposerMeta := numNodesPerShard
```
Comment (bogdan-rosianu) [inline integrationTests/sync/basicSync_test.go]: move this line above so you can use idxProposerMeta on line 46

File: integrationTests/testSyncNode.go [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+	"fmt"
+
+	"github.com/ElrondNetwork/elrond-go/consensus/spos/sposFactory"
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/integrationTests/mock"
+	"github.com/ElrondNetwork/elrond-go/node"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/process/block"
+	"github.com/ElrondNetwork/elrond-go/process/smartContract"
+	processSync "github.com/ElrondNetwork/elrond-go/process/sync"
```
Comment (BeniaminDrasovean) [inline integrationTests/testSyncNode.go]: Is the processSync nickname really needed?

File: integrationTests/singleShard/transaction/interceptedBulkTx_test.go [hunk 4]
```diff
@@ -103,3 +107,72 @@ func TestNode_GenerateSendInterceptBulkTransactionsWithMessenger(t *testing.T) {
...
+	numOfNodes := 4
+	idxProposer := 0
+	mintingValue := big.NewInt(100000000)
+	advertiser := integrationTests.CreateMessengerWithKadDht(context.Background(), "")
+	_ = advertiser.Bootstrap()
+	advertiserAddr := integrationTests.GetConnectableAddress(advertiser)
+
+	nodes := make([]*integrationTests.TestProcessorNode, numOfNodes)
+	for i := 0; i < numOfNodes; i++ {
+		nodes[i] = integrationTests.NewTestProcessorNode(maxShards, 0, 0, advertiserAddr)
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply integrationTests/singleShard/transaction/interceptedBulkTx_test.go]: Removed test. This was not expected to get in this PR.

File: integrationTests/sync/basicSync_test.go (hunk 1 above)
Comment (iulianpascalau) [reply integrationTests/sync/basicSync_test.go]: In tests we do not treat each and every error (for that purpose we have unit tests). If it happens for an error to be raised here, the test will certainly fail.

File: integrationTests/singleShard/transaction/interceptedBulkTx_test.go [hunk 5]
```diff
@@ -1,20 +1,24 @@
...
 
+	"github.com/ElrondNetwork/elrond-go/crypto"
 	"github.com/ElrondNetwork/elrond-go/data"
 	"github.com/ElrondNetwork/elrond-go/data/transaction"
 	"github.com/ElrondNetwork/elrond-go/integrationTests"
 	"github.com/ElrondNetwork/elrond-go/process"
 	"github.com/stretchr/testify/assert"
 )
 
+var stepDelay = time.Second
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply integrationTests/singleShard/transaction/interceptedBulkTx_test.go]: ok

File: integrationTests/testSyncNode.go (hunk 3 above)
Comment (iulianpascalau) [reply integrationTests/testSyncNode.go]: no, was copy-pasted. Will remove alias

File: integrationTests/sync/basicSync_test.go (hunk 2 above)
Comment (iulianpascalau) [reply integrationTests/sync/basicSync_test.go]: done

File: integrationTests/singleShard/transaction/interceptedBulkTx_test.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply integrationTests/singleShard/transaction/interceptedBulkTx_test.go]: done

File: integrationTests/singleShard/transaction/interceptedBulkTx_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply integrationTests/singleShard/transaction/interceptedBulkTx_test.go]: fair enough

**Pull request merged**